accession-commons-core/src/main/java/uk/ac/ebi/ampt2d/commons/accession/core/aop/DecorateAccessionAfterThrowingAccessionMergedException.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2019 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package uk.ac.ebi.ampt2d.commons.accession.core.aop;
+
+import org.aspectj.lang.JoinPoint;
+import org.aspectj.lang.annotation.AfterThrowing;
+import org.aspectj.lang.annotation.Aspect;
+import uk.ac.ebi.ampt2d.commons.accession.core.DecoratedAccessioningService;
+import uk.ac.ebi.ampt2d.commons.accession.core.exceptions.AccessionMergedException;
+
+/**
+ * An @Aspect for ensuring accessions are decorated in AccessionMergedExceptions
+ */
+@Aspect
+public class DecorateAccessionAfterThrowingAccessionMergedException {
```

Comment:
`DecoratedAccessionMergedExceptionAspect`? Looking at the original class name it sounds like an `exception` class because the name ends with `Exception`. Also I don't think you need to say `AfterThrowing` in the class name.